lib/db/reasoning.js

```diff
@@ -0,0 +1,117 @@
+const { fetch } = require('../fetch');
+
+const { httpBody } = require('../response-transforms');
+
+const jsonify = res => {
+  res.headers.append('content-type', 'application/json');
+  return res;
+};
+
+const consistency = (conn, database, options, params) => {
+  const headers = conn.headers();
+  headers.append('Accept', 'text/boolean');
+
+  const uri = `${conn.uri(
+    database,
+    'reasoning',
+    'consistency'
+  )}${options.namedGraph ? `?graph-uri=${options.namedGraph}` : ''}`;
+
+  return fetch(uri, {
+    headers,
+  }).then(httpBody);
+};
+
+// contentType - application/x-turtle, text/turtle, application/rdf+xml, text/plain, application/x-trig, text/x-nquads, application/trix
+const explainInference = (conn, database, inference, options, params = {}) => {
```

Comment:
So since `options` is required here, because you need to specify contentType, I've been calling those kinds of arguments, `config`.